src/overlays/actors/ovl_Elf_Msg3/z_elf_msg3.c

```diff
@@ -29,28 +29,121 @@ const ActorInit Elf_Msg3_InitVars = {
     (ActorFunc)NULL,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80A2D1E0[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneForward, 1000, ICHAIN_STOP),
 };
 
-#endif
-
-extern InitChainEntry D_80A2D1E0[];
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/D_80A2D1F0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/ElfMsg3_SetupAction.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/func_80A2CD1C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/ElfMsg3_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/ElfMsg3_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/func_80A2CF50.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/func_80A2CF7C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Elf_Msg3/ElfMsg3_Update.s")
+void ElfMsg3_SetupAction(ElfMsg3* this, ElfMsg3ActionFunc actionFunc) {
+    (void)"共倒れ"; // "Collapse together"
+    (void)"共倒れ";
+    this->actionFunc = actionFunc;
+}
+
+s32 func_80A2CD1C(ElfMsg3* this, PlayState* play) {
+    if ((this->actor.home.rot.y > 0) && (this->actor.home.rot.y < 0x81) &&
+        (Flags_GetSwitch(play, this->actor.home.rot.y - 1))) {
+        if (ELFMSG3_GET_SWITCH(&this->actor) != 0x7F) {
+            Flags_SetSwitch(play, ELFMSG3_GET_SWITCH(&this->actor));
+        }
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    if (this->actor.home.rot.y == 0x81) {
+        if (Flags_GetClear(play, this->actor.room)) {
+            if (ELFMSG3_GET_SWITCH(&this->actor) != 0x7F) {
+                Flags_SetSwitch(play, ELFMSG3_GET_SWITCH(&this->actor));
+            }
+            Actor_MarkForDeath(&this->actor);
+            return true;
+        }
+    }
+    if (ELFMSG3_GET_SWITCH(&this->actor) == 0x7F) {
+        return false;
+    }
+    if (Flags_GetSwitch(play, ELFMSG3_GET_SWITCH(&this->actor))) {
+        Actor_MarkForDeath(&this->actor);
+        return true;
+    }
+    return false;
+}
+
+void ElfMsg3_Init(Actor* thisx, PlayState* play) {
+    ElfMsg3* this = THIS;
+    s32 baseRange;
+
+    if (func_80A2CD1C(this, play) == 0) {
+        Actor_ProcessInitChain(&this->actor, sInitChain);
+        if (ABS_ALT(this->actor.home.rot.x) == 0) {
+            this->actor.scale.z = 0.4f;
+            this->actor.scale.x = 0.4f;
+        } else {
+            baseRange = ABS_ALT(this->actor.home.rot.x);
+            this->actor.scale.z = baseRange * 0.04f;
+            this->actor.scale.x = baseRange * 0.04f;
```

Comment:
```suggestion
            this->actor.scale.x = this->actor.scale.z= ABS_ALT(this->actor.home.rot.x) & 0.04f;
```
Should work and would allow you to remove the `baseRange` variable.